src/auth/decorators/current-user-id.decorator.ts

```diff
@@ -0,0 +1,10 @@
+import { ExecutionContext, createParamDecorator } from '@nestjs/common';
+
+import { Request } from 'express';
+
+export const CurrentUserId = createParamDecorator(
+  (_: undefined, context: ExecutionContext) => {
+    const request: Request = context.switchToHttp().getRequest();
```

Comment:
**suggestion:** there is no need to declare the module in the other file, you can just create an interface which extends the Request and use the generic template offered by the `getRequest<T: any>() : T` method